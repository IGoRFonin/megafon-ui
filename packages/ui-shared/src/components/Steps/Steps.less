@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-beta-steps');

@{b} {
    width: 51%;
    margin-left: auto;
    margin-right: auto;

    @media @desktopS {
        width: 68%;
    }

    @media @mobileB {
        width: 74%;
    }

    @media @mobileSM {
        width: 100%;
    }

    &__title {
        margin-bottom: 40px;
    }

    &__list {
        list-style: none;
        margin: 0;
        padding: 0;
    }

    &__item {
        display: flex;
        margin-bottom: 48px;

        @media @mobileSM {
            margin-bottom: 24px;
        }

        &:last-child {
            margin-bottom: 0;
        }
    }

    &__step-number {
        .h5();
        display: flex;
        align-items: center;
        justify-content: center;
        flex-shrink: 0;
        width: 40px;
        height: 40px;
        border-radius: 50%;
        background-color: @green;
        color: @clearWhite;
    }

    &__text-wrapper {
        padding-top: 8px;
        padding-left: 16px;
    }
}
