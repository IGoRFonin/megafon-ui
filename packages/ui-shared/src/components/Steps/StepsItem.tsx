import React from 'react';
import PropTypes from 'prop-types';
import { cnCreate, convert, Paragraph, TextLink } from '@megafon/ui-core';
import './StepsItem.less';

const convertConfig = {
    a: {
        component: TextLink,
        props: ['href', 'target'],
    },
    b: {
        component: ({ children }) => <b>{children}</b>,
    },
};

interface IStepsItem {
    /** Номер элемента */
    index: number;
    /** Текст */
    text: string;
}

const cn = cnCreate('mfui-beta-steps-item');
const StepsItem: React.FC<IStepsItem> = ({ index, text }) => (
    <div className={cn()}>
        <span className={cn('step-number')}>
            {index}
        </span>
        <div className={cn('text-wrapper')}>
            <Paragraph hasMargin={false}>{convert(text, convertConfig)}</Paragraph>
        </div>
    </div>
);

StepsItem.propTypes = {
    index: PropTypes.number.isRequired,
    text: PropTypes.string.isRequired,
};

export default StepsItem;
