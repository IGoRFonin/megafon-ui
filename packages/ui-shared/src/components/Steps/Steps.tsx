import * as React from 'react';
import { cnCreate, Paragraph, convert, TextLink, Header } from '@megafon/ui-core';
import './Steps.less';

const convertConfig = {
    a: {
        component: TextLink,
        props: ['href', 'target'],
    },
    b: {
        component: ({ children }) => <b>{children}</b>,
    },
};

export type ItemType = {
    text: string;
};

export interface ISteps {
    /** Заголовок */
    title: string;
    /** Данные для списка шагов */
    items: ItemType[];
}

const cn = cnCreate('mfui-beta-steps');
const Steps: React.FC<ISteps> = ({ items, title }) => (
    <div className={cn()}>
        <Header as="h2" hAlign="center" className={cn('title')}>{title}</Header>
        <ul className={cn('list')}>
            {items.map(({ text }, index) => (
                <li key={index} className={cn('item')}>
                    <span className={cn('step-number')}>
                        {index + 1}
                    </span>
                    <div className={cn('text-wrapper')}>
                        <Paragraph hasMargin={false}>{convert(text, convertConfig)}</Paragraph>
                    </div>
                </li>
            ))}
        </ul>
    </div>
);

export default Steps;
