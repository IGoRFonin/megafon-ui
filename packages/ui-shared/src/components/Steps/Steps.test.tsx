import * as React from 'react';
import { shallow } from 'enzyme';
import Steps, { ISteps } from './Steps';

const props: ISteps = {
    title: 'title',
    items: [
        {
            text: 'text-1',
        },
        {
            text: 'text with <a href="#" target="_self">link</a>',
        },
        {
            text: 'bold <b>text</b>',
        },
    ],
};

describe('Steps', () => {
    it('should render component', () => {
        const wrapper = shallow(<Steps {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
