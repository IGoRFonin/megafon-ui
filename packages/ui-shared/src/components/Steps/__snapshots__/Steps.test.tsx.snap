// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps should render component 1`] = `
<div
  className="mfui-beta-steps"
>
  <Header
    as="h2"
    className="mfui-beta-steps__title"
    color="black"
    hAlign="center"
  >
    title
  </Header>
  <ul
    className="mfui-beta-steps__list"
  >
    <li
      className="mfui-beta-steps__item"
      key="0"
    >
      <span
        className="mfui-beta-steps__step-number"
      >
        1
      </span>
      <div
        className="mfui-beta-steps__text-wrapper"
      >
        <Paragraph
          hasMargin={false}
        >
          text-1
        </Paragraph>
      </div>
    </li>
    <li
      className="mfui-beta-steps__item"
      key="1"
    >
      <span
        className="mfui-beta-steps__step-number"
      >
        2
      </span>
      <div
        className="mfui-beta-steps__text-wrapper"
      >
        <Paragraph
          hasMargin={false}
        >
          text with 
          <TextLink
            color="blue"
            href="#"
            key="1"
            target="_self"
            underlineStyle="solid"
            underlineVisibility="hover"
          >
            link
          </TextLink>
        </Paragraph>
      </div>
    </li>
    <li
      className="mfui-beta-steps__item"
      key="2"
    >
      <span
        className="mfui-beta-steps__step-number"
      >
        3
      </span>
      <div
        className="mfui-beta-steps__text-wrapper"
      >
        <Paragraph
          hasMargin={false}
        >
          bold 
          <component
            key="1"
          >
            text
          </component>
        </Paragraph>
      </div>
    </li>
  </ul>
</div>
`;
